analyzer/src/traversal/contracts.rs

```diff
@@ -29,14 +29,17 @@ pub fn contract_def(
     module_scope: Shared<ModuleScope>,
     context: Shared<Context>,
     stmt: &Spanned<fe::ModuleStmt>,
-) -> Result<(), SemanticError> {
+) -> Result<Shared<ContractScope>, SemanticError> {
     if let fe::ModuleStmt::ContractDef { name, body } = &stmt.node {
-        let contract_scope = ContractScope::new(Rc::clone(&module_scope));
+        let contract_scope = ContractScope::new(name.node, Rc::clone(&module_scope));
 
         for stmt in body.iter() {
             match &stmt.node {
                 fe::ContractStmt::ContractField { .. } => {
-                    contract_field(Rc::clone(&contract_scope), stmt)
+                    // Contract fields are evaluated in the next pass together with function bodies
+                    // so that they can use other contract types that may only be defined after the
+                    // current contract.
+                    Ok(())
```

Comment:
@g-r-a-n-t since fields aren't in the public API of contracts it omitting them here doesn't interfere with the contract representation in the scope. But by delaying analysis to the next pass we can at least support referring to subsequent contracts in contract fields.